### PR TITLE
chore: replace old mui itemlist selector with ui-core components

### PR DESCRIPTION
**Pull request by jenniferarnesen**
These changes affect dashboard edit mode, including the name, description fields and the Item Selector. All types and functionalities should be checked in the item selector. The styles are updated to reflect the latest sketches here:
* https://sketch.cloud/s/A0qyy/a/dqe0pL/play
* https://codepen.io/j_cpr/live/73918a10a96df4de0acaeec4e023edcf 

There are a few workarounds in place until the ui-core textarea is ready: 
1.  src/index.js - overriding the palette color - this is to get the same label/underline color in the description field as the title field. I decided not to change the theme in d2-ui, since the whole thing is probably out-of-date and needs to be dealt with separately.
2. EditTitleBar.js - override the border-bottom style for the pseudo-element :before. This is to have the same underlining behaviour as the Title field, which uses a ui-core TextField.

### CODE REVIEW

**Comment by cooper-joe** [comment]
Feedback/requested changes:
- 'Show more' sometimes shows nothing more. For example 'Apps' shows just 'Social Media Video' as the only item, but there is still a 'show more' control. Clicking it does nothing except change control text to 'show fewer'. This control should be hidden if there are =<5.
- add an active (click) state to the items, [in my codepen](https://codepen.io/j_cpr/live/73918a10a96df4de0acaeec4e023edcf) I added `background-color:grey400;`

This looks and works great, thanks!

**Review by cooper-joe — CHANGES_REQUESTED** [review]
Feedback/requested changes:

- 'Show more' sometimes shows nothing more. For example 'Apps' shows just 'Social Media Video' as the only item, but there is still a 'show more' control. Clicking it does nothing except change control text to 'show fewer'. This control should be hidden if there are =<5.
- add an active (click) state to the items, in [the codepen](https://codepen.io/j_cpr/live/73918a10a96df4de0acaeec4e023edcf) I added `background-color:grey400;`

This looks and works great, thanks!

File: src/components/ItemSelector/ItemSelector.js [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+            anchorEl: event.currentTarget,
+            open: true,
+        });
+    };
+
+    setFilter = event => {
+        this.setState({ filter: event.target.value }, this.fetchItems);
+    };
+
+    getCatetorizedMenuGroups = items =>
```
Comment (edoardo) [inline src/components/ItemSelector/ItemSelector.js]: Typo.

**Review by edoardo — APPROVED** [review]
LGTM

**Review by cooper-joe — APPROVED** [review]
approved, the issues I raised will be handled elsewhere:
- show more/show less: backend change requested, so updates will be handled in another pull request
- ui-core menus to be updated with an active state

**Pull request merged**